### PR TITLE
native: create & join followup

### PR DESCRIPTION
**Pull request by latter-bolden**
WIP

Fixes TLON-1911
Fixes TLON-1914
Fixes TLON-1916
Fixes TLON-1918
Fixes TLON-1919
Fixes TLON-1921

TODO:
- [ ] Chat list perf (sheets heavy even when closed?)

### CODE REVIEW

**Comment by latter-bolden** [comment]
After talking with @dnbrwstr, we're going to prioritize getting this in to avoid refactor collisions. All TODOS apart from `ChatListScreen` have been split out (see TLON-1936 and TLON-1935)

**Comment by dnbrwstr** [comment]
(Also, I took on cleaning up + landing this). @latter-bolden: should be good to review when you're ready.

File: apps/tlon-mobile/src/screens/ChannelScreen.tsx [hunk 1]
```diff
@@ -225,7 +225,9 @@ export default function ChannelScreen(props: ChannelScreenProps) {
   const handleGroupAction = useCallback(
     async (action: GroupPreviewAction, updatedGroup: db.Group) => {
       if (action === 'goTo' && updatedGroup.lastPost?.channelId) {
-        const channel = await store.getChannel(updatedGroup.lastPost.channelId);
+        const channel = await db.getChannel({
```
Comment (latter-bolden) [inline apps/tlon-mobile/src/screens/ChannelScreen.tsx:228]: a `db` method called from a screen component *gasp*.  Not a huge deal, but feels weird to me

**Review by latter-bolden — COMMENTED** [review]
Looking good! ✅🌿🦖🥬

File: apps/tlon-mobile/src/screens/ChannelScreen.tsx (hunk 1 above)
Comment (dnbrwstr) [reply apps/tlon-mobile/src/screens/ChannelScreen.tsx:228]: Lol yaa there are a few of these -- I think store should re-export queries ultimately, but that's for a later pass.

**Pull request merged**